erica_app/erica/elster_xml/xml_parsing/erica_xml_parsing.py

```diff
@@ -10,40 +10,33 @@ def _get_element_from_xml(xml_string: str, element_xpath: str):
     return element.text
 
 
-def get_antrag_id_from_xml(xml_string):
-    return _get_element_from_xml(xml_string, 'AntragsID')
-
-
-def get_idnr_from_xml(xml_string):
-    return _get_element_from_xml(xml_string, 'DateninhaberIdNr')
-
-
-def get_transfer_ticket_from_xml(xml_string):
-    return _get_element_from_xml(xml_string, 'TransferTicket')
-
-
-def get_address_from_xml(xml_string):
-    return _get_element_from_xml(xml_string, 'AdrKette')
-
-
-def get_relevant_beleg_ids(xml_string, beleg_types):
-    beleg_id_xml = remove_declaration_and_namespace(xml_string)
-    beleg_id_elements = beleg_id_xml.findall('.//Id')
-    beleg_ids = []
-    for beleg_id_element in beleg_id_elements:
-        belegart = beleg_id_element.get('belegart')
-        if belegart in beleg_types:
-            beleg_ids.append(beleg_id_element.text.strip())
-
-    return beleg_ids
-
-
 def remove_declaration_and_namespace(xml_string):
     import xml.etree.ElementTree as ET
     xml_string = re.sub(' xmlns="[^"]+"', '', xml_string, count=1)
     return ET.fromstring(xml_string)
 
 
+def get_elements_from_xml(xml_string, element):
+    xml_tree = remove_declaration_and_namespace(xml_string)
+    matched_elements = xml_tree.findall('.//' + element)
+    result = []
+    for el in matched_elements:
+        result.append(el)
+    if xml_tree.tag == element:  # also consider the element itself
+        result.append(xml_tree)
+    return result
+
+
+def get_elements_text_from_xml_element(xml_element, searched_child_element):
+    matched_elements = xml_element.findall('.//' + searched_child_element)
+    result = []
+    for el in matched_elements:
+        result.append(el.text)
+    if xml_element.tag == searched_child_element:  # also consider the element itself
+        result.append(xml_element.text)
+    return result
+
+
```

Comment:
I like this, but there's a lot of similar logic in these two functions (and also get_elements_text_from_xml). Maybe we could have a function that does `get_elements_from_xml_element` and the others just call that?. It's more looping but also clearer code. WDYT?

Something like this (untested):
```suggestion
def get_elements_from_xml_element(xml_element, tag_name):
    matched_elements = xml_element.findall('.//' + tag_name)
    result = []
    for el in matched_elements:
        result.append(el)
    if xml_element.tag == element:  # also consider the element itself
        result.append(xml_element)
    return result
    
    
def get_elements_from_xml(xml_string, element):
    xml_tree = remove_declaration_and_namespace(xml_string)
    return get_elements_from_xml_element(xml_tree, element)


def get_elements_text_from_xml_element(xml_element, searched_child_element):
    return [el.text for el in get_elements_from_xml_element(xml_element, searched_child_element)


def get_elements_text_from_xml(xml_string, element):
    xml_tree = remove_declaration_and_namespace(xml_string)
    return [el.text for el in get_elements_from_xml_element(xml_tree, searched_child_element)
```